app/controllers/schools_controller.rb

```diff
@@ -101,8 +110,40 @@ def destroy
 
 private
 
+  def user_signed_in_and_linked_to_school?
+    user_signed_in? && (current_user.school_id == @school.id)
+  end
+
+  def not_signed_in?
+    !user_signed_in? || current_user.guest?
+  end
+
+  def redirect_if_not_visible
+    redirect_to school_inactive_path(@school) if user_signed_in_and_linked_to_school? && !@school.visible?
+  end
+
+  def redirect_pupils
+    redirect_to pupils_school_path(@school) if user_signed_in_and_linked_to_school? && current_user.pupil? && !switch_dashboard?
+  end
+
+  def switch_dashboard?
+    params[:switch].present? && params[:switch] == "true"
+  end
+
+  def redirect_to_pupil_dash_if_not_data_enabled
+    redirect_to pupils_school_path(@school) if not_signed_in? && !@school.data_enabled
+  end
+
   def setup_default_features
     @observations = setup_timeline(@school.observations)
+    #Setup management dashboard features if users has permission
+    #to do that
+    if can?(:show_management_dash, @school)
+      @add_contacts = site_settings.message_for_no_contacts && @school.contacts.empty? && can?(:manage, Contact)
+      @add_pupils = site_settings.message_for_no_pupil_accounts && @school.users.pupil.empty? && can?(:manage_users, @school)
+      @prompt_training = !@show_data_enabled_features || current_user.confirmed_at < 60.days.ago
+      @prompt_for_bill = @school.bill_requested && can?(:index, ConsentDocument)
```

Comment:
perhaps "prompt training" and `prompt for bill` could be methods on the user model? something like
```
PROMPT_TRAINING_DAY_THRESHOLD = 60
...
def prompt_training?
  confirmed_at && (confirmed_at < PROMPT_TRAINING_DAY_THRESHOLD.days.ago)
end
```

```
def prompt_for_bill?
  return unless can?(:index, ConsentDocument)
  school.bill_requested
end
```